### PR TITLE
ref(ts): convert ProjectPlugins to TS

### PR DESCRIPTION
**Pull request by scefali**

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/projectPlugins/projectPlugins.tsx [hunk 1]
```diff
@@ -68,9 +78,6 @@ class ProjectPlugins extends Component {
 
           {plugins
             .filter(p => {
-              if (DEPRECATED_PLUGINS.includes(p.id) && !p.enabled) {
```
Comment (scefali) [inline src/sentry/static/sentry/app/views/settings/projectPlugins/projectPlugins.tsx:71]: The deprecated plugins of `taiga`, `lighthouse`, and `grove-io` have been long removed so this code is unnecessary.

Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/projectPlugins/projectPlugins.tsx:71]: Can we remove `DEPRECATED_PLUGINS` elsewhere too?

File: src/sentry/static/sentry/app/views/settings/projectPlugins/projectPlugins.tsx [hunk 2]
```diff
@@ -14,11 +15,20 @@ import Link from 'app/components/links/link';
...
 
-class ProjectPlugins extends Component {
+type Props = {
+  plugins: Plugin[];
+  loading: boolean;
+  error: React.ComponentProps<typeof RouteError>['error'];
+  onChange: React.ComponentProps<typeof ProjectPluginRow>['onChange'];
+} & WithRouterProps<{orgId: string}>;
+
+type State = {};
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/projectPlugins/projectPlugins.tsx:29]: is State needed?

File: src/sentry/static/sentry/app/views/settings/projectPlugins/projectPlugins.tsx (hunk 1 above)
Comment (scefali) [reply src/sentry/static/sentry/app/views/settings/projectPlugins/projectPlugins.tsx:71]: @billyvg done

Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/projectPlugins/projectPlugins.tsx:71]: Nice

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/views/settings/projectPlugins/details.jsx [hunk 3]
```diff
@@ -154,17 +152,9 @@ class ProjectPluginDetails extends AsyncView {
...
-    const disabled = DEPRECATED_PLUGINS.includes(pluginId);
 
     const enable = (
-      <Button
-        disabled={disabled}
-        size="small"
-        onClick={this.handleEnable}
-        style={{marginRight: '6px'}}
-      >
+      <Button size="small" onClick={this.handleEnable} style={{marginRight: '6px'}}>
```
Comment (matejminar) [inline src/sentry/static/sentry/app/views/settings/projectPlugins/details.jsx:157]: NIT: We could use a `space` util here.

**Review by matejminar — APPROVED** [review]

File: src/sentry/static/sentry/app/views/settings/projectPlugins/details.jsx (hunk 3 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/views/settings/projectPlugins/details.jsx:157]: and emotion 🙃

Comment (scefali) [reply src/sentry/static/sentry/app/views/settings/projectPlugins/details.jsx:157]: @matejminar @priscilawebdev good suggestions, I'll remember to make those changes when I convert this file in the future.

**Pull request merged**